pandas/core/generic.py

```diff
@@ -9964,11 +9978,24 @@ def _check_percentile(self, q):
         """
 
     @Appender(_shared_docs['pct_change'] % _shared_doc_kwargs)
-    def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
-                   **kwargs):
+    def pct_change(self, skipna=None, periods=1, fill_method=None, limit=None,
+                   freq=None, **kwargs):
+        if skipna and fill_method is not None:
+            raise ValueError("cannot pass both skipna and fill_method")
+        elif skipna and limit is not None:
+            raise ValueError("cannot pass both skipna and limit")
+        if skipna is None and fill_method is None and limit is None:
+            skipna = True
+        if skipna and isinstance(self, pd.DataFrame):
```

Comment:
why is the skipna here? this is strange why you are now handling DataFrame different than before

Reply:
Yes, DataFrames are handled differently. The problem is that for a DataFrame, the NAs must be dropped independently for each column/row, and this is not possible with `DataFrame.dropna` (thus the need of using `DataFrame.apply` instead). See my question (https://github.com/pandas-dev/pandas/issues/25006#issuecomment-462153438) and the answer by @WillAyd (https://github.com/pandas-dev/pandas/issues/25006#issuecomment-462198723) in the issue #25006.

Reply:
I am generally not a fan of introspection in a shared method - any reason you wouldn't subclass this for DataFrame as required?